explore_RNAseq_only_generic_genes/1c_original_vs_simulated_counts.py

```diff
@@ -0,0 +1,227 @@
+            # Get simulated filename
+            simulated_filename = os.path.join(
+                simulated_dir, f"selected_simulated_data_{project_id}_{i}_processed.txt"
+            )
+
+            # Read simulated experiment
+            simulated = pd.read_csv(simulated_filename, sep="\t", index_col=0, header=0)
+
+            # Create dataframe with columns
+            # Sample id | total counts across genes original (template)| total counts across genes after VAE (simulated)
+            template_sample_counts = template.sum(axis=1)
+            simulated_sample_counts = simulated.sum(axis=1)
+
+            sample_counts = template_sample_counts.to_frame("Original counts").merge(
+                simulated_sample_counts.to_frame("Simulated counts"),
+                left_index=True,
+                right_index=True,
+            )
+
+            # Plot per sample
+            f = sns.scatterplot(
+                data=sample_counts,
+                x="Simulated counts",
+                y="Original counts",
+                alpha=0.8,
+                ax=axes[i],
+            )
+
+            min_ = sample_counts.min().min()
+            max_ = sample_counts.max().max()
+            f.plot([min_, max_], [min_, max_], "k--")
+
+            axes[i].set_ylabel("")
+            axes[i].set_xlabel("")
+
+        fig.text(
+            0.5,
+            0.0,
+            "Simulated counts (after VAE)",
+            ha="center",
+            fontsize=14,
+            fontname="Verdana",
+        )
+        fig.text(
+            0.08,
+            0.5,
+            "Actual counts",
+            va="center",
+            rotation="vertical",
+            fontsize=14,
+            fontname="Verdana",
+        )
+        fig.suptitle(
+            "Actual vs simulated counts per sample",
+            fontsize=16,
+            fontname="Verdana",
+        )
+    elif by_sample_or_gene == "gene":
+        for i in range(num_simulated):
+            # Get simulated filename
+            simulated_filename = os.path.join(
+                simulated_dir, f"selected_simulated_data_{project_id}_{i}_processed.txt"
+            )
+
+            # Read simulated experiment
+            simulated = pd.read_csv(simulated_filename, sep="\t", index_col=0, header=0)
+
+            # Create dataframe with columns
+            # Gene id | total counts across samples original (template)| total counts across samples after VAE (simulated)
+            template_gene_counts = template.sum()
+            simulated_gene_counts = simulated.sum()
+            # template_gene_counts = template.iloc[1]
+            # simulated_gene_counts = simulated.iloc[1]
+
+            gene_counts = template_gene_counts.to_frame("Original counts").merge(
+                simulated_gene_counts.to_frame("Simulated counts"),
+                left_index=True,
+                right_index=True,
+            )
+
+            # Plot per sample
+            g = sns.scatterplot(
```

Comment:
So each point on this plot is a gene, and each subplot is a sample?

How did you choose the genes to look at here? Are these genes that are generic in RNA-seq but not array, and lowly expressed in the RNA-seq data?

Reply:
So these are total counts summed across all genes, to get a sense for the sequencing coverage per sample that will affect the DESeq, which will give more variance (probability of error) to those genes.

https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-10-r106

Reply:
I'll update the text in the notebook so this is clear. Its a good question